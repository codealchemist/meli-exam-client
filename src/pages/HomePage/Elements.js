import styled from 'styled-components'

export const Content = styled.div`
  width: 100%;
  overflow: hidden;

  img {
    margin: -2px 0;
  }
`

export const CenteredImage = styled.div`
  width: 100%;
  height: ${props => props.height || '100%'};
  background: url(${props => props.src}) no-repeat;
  background-size: cover;
  background-position: center;
`
