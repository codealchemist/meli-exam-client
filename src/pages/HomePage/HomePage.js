import React from 'react'
import { img1, img2, img3, img4 } from './Images'
import { Content, CenteredImage } from './Elements'

const HomePage = () => (
  <div>
    <Content>
      <CenteredImage src={img1} height="30vw" />
      <CenteredImage src={img2} height="30vw" />
      <CenteredImage src={img3} height="30vw" />
      <CenteredImage src={img4} height="30vw" />
    </Content>
  </div>
)

export default HomePage
