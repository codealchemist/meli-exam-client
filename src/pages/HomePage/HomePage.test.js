import React from 'react'
import { mount } from 'enzyme'
import HomePage from './HomePage'

describe('HomePage', () => {
  it('should render', () => {
    const wrapper = mount(<HomePage />)
    expect(wrapper).toMatchSnapshot()
  })
})
