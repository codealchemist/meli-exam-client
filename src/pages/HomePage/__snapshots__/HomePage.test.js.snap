// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render 1`] = `
<HomePage>
  <div>
    <styled.div>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "lastClassName": "elYfoL",
              "rules": Array [
                "
  width: 100%;
  overflow: hidden;

  img {
    margin: -2px 0;
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="sc-bdVaJa elYfoL"
        >
          <styled.div
            height="30vw"
            src="img1.jpg"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "lgGhRR",
                    "rules": Array [
                      "
  width: 100%;
  height: ",
                      [Function],
                      ";
  background: url(",
                      [Function],
                      ") no-repeat;
  background-size: cover;
  background-position: center;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              height="30vw"
              src="img1.jpg"
            >
              <div
                className="sc-bwzfXH cCHmzG"
                height="30vw"
                src="img1.jpg"
              />
            </StyledComponent>
          </styled.div>
          <styled.div
            height="30vw"
            src="img2.jpg"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "lgGhRR",
                    "rules": Array [
                      "
  width: 100%;
  height: ",
                      [Function],
                      ";
  background: url(",
                      [Function],
                      ") no-repeat;
  background-size: cover;
  background-position: center;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              height="30vw"
              src="img2.jpg"
            >
              <div
                className="sc-bwzfXH jTaqDS"
                height="30vw"
                src="img2.jpg"
              />
            </StyledComponent>
          </styled.div>
          <styled.div
            height="30vw"
            src="img3.jpg"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "lgGhRR",
                    "rules": Array [
                      "
  width: 100%;
  height: ",
                      [Function],
                      ";
  background: url(",
                      [Function],
                      ") no-repeat;
  background-size: cover;
  background-position: center;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              height="30vw"
              src="img3.jpg"
            >
              <div
                className="sc-bwzfXH bWXqaU"
                height="30vw"
                src="img3.jpg"
              />
            </StyledComponent>
          </styled.div>
          <styled.div
            height="30vw"
            src="img4.jpg"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "lgGhRR",
                    "rules": Array [
                      "
  width: 100%;
  height: ",
                      [Function],
                      ";
  background: url(",
                      [Function],
                      ") no-repeat;
  background-size: cover;
  background-position: center;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              height="30vw"
              src="img4.jpg"
            >
              <div
                className="sc-bwzfXH lgGhRR"
                height="30vw"
                src="img4.jpg"
              />
            </StyledComponent>
          </styled.div>
        </div>
      </StyledComponent>
    </styled.div>
  </div>
</HomePage>
`;
